Testnet/Airdrop/Airdrop.Tests/AirdropTests.cs

```diff
@@ -0,0 +1,814 @@
+            Assert.Equal(expectedAmountToDistribute, amountToDistribute);
+        }
+        #endregion
+
+        #region Helpers
+
+        private class TransferResult : ITransferResult
+        {
+            public TransferResult(bool success = true)
+            {
+                Success = success;
+            }
+
+            public object ReturnValue => null;
+            public bool Success { get; set; }
+        }
+
+        private struct MockWithdrawExecuteParams
+        {
+            public Address Sender;
+            public Airdrop Airdrop;
+            public Status Status;
+            public ulong ExpectedAmountToDistribute;
+            public TransferResult TransferResult;
+        }
+
+        /// <summary>
+        /// Initializes an aidrop instance and sets mocks accordingly to interact with Contract and Persistent State.
+        /// </summary>
+        /// <param name="sender">Address of the sender of the message</param>
+        /// <param name="owner">Owner of the contract</param>
+        /// <param name="currentBlock">CurrentBlock transaction will run on</param>
+        /// <param name="endBlock">Endblock of the airdrop registration period</param>
+        /// <param name="totalSupply">TotalSupply that will be airdropped</param>
+        /// <returns>Airdrop instance</returns>
+        private Airdrop InitializeTest(Address sender, Address owner, ulong currentBlock, ulong endBlock, ulong totalSupply)
+        {
+            // Set the new call from the sender to the contract
+            MockContractState.Setup(x => x.Message).Returns(new Message(AirdropContractAddress, sender, 0));
+            // Block.Number returns currentBlock supplied by param
+            MockContractState.Setup(b => b.Block.Number).Returns(currentBlock);
+            // EndBlock returns endblock supplied by param
+            MockPersistentState.Setup(x => x.GetUInt64(nameof(EndBlock))).Returns(endBlock);
+            // TotalSupply returns totalSupply supplied by param
+            MockPersistentState.Setup(x => x.GetUInt64(nameof(TotalSupply))).Returns(totalSupply);
+            // Owner returns owner supplied by param
+            MockPersistentState.Setup(x => x.GetAddress(nameof(Owner))).Returns(owner);
+            // TokenContractAddress returns tokenContractAddress supplied by param
+            MockPersistentState.Setup(x => x.GetAddress(nameof(TokenContractAddress))).Returns(TokenContractAddress);
+
+            // NumberOfRegistrants returns NumberOfRegistrants from this test class
+            MockPersistentState.Setup(x => x.GetUInt64(nameof(NumberOfRegistrants))).Returns(NumberOfRegistrants);
+            // NumberOfRegistrants, when set, increments the NumberOfRegistrations in this test class
+            MockPersistentState.Setup(x => x.SetUInt64(nameof(NumberOfRegistrants), It.IsAny<ulong>()))
+                .Callback<string, ulong>((key, value) => { NumberOfRegistrants += value; });
```

Comment:
Mocking the contract's implementation in this way will prevent any problems with the actual implementation being found. I'd recommend just setting `NumberOfRegistrants = value` here, and slightly modifying the test that the value is incremented.